controller/service.go

```diff
@@ -123,6 +136,25 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 			c.clearServiceState(key, svc)
 			lbIPs = []net.IP{}
 		}
+		if len(desiredLbIPs) > 0 {
+			desiredPool, err = c.ips.GetPoolOwner(desiredLbIPs)
```

Comment:
as I wrote below, I feel this logic belongs more to the allocator. Basically what we need to ask is, given a pool name, is it compatible with my service?